### ISSUE TITLE
Ensure good out-of-box experience when there are network issues

### ISSUE THREAD

**Issue by gorhill** [uBlock-LLC/uBlock#426]
uBlock comes with _"Auto-update"_ enabled.

This means that first time uBlock is launched, it will fetch resources (EasyList, EasyPrivacy, Fanboy's Social, Peter Lowe's) which have a remote "home" from their respective remote servers.

This means if there happens to be any network- or remote server-related issue when uBlock launches, the attempts to fetch remotely will timeout (15s-30s), causing a significant delay for when uBlock is ready to actually perform its job. A user do not expect that delay, and will try right away whether uBlock does block as advertised, and will find out it doesn't if all the timeouts (due to network issues) have not been fired yet (when they are uBlock will fall back to local copies shipped with the extensions, which are likely out-dated however).

In short, when there are network issues beyond the control of uBlock, an out-of-the-box experience risks not being positive.

Example: https://archive.rebeccablacktech.com/g/thread/S45680589#p45697232

Solution: Absolutely avoid completely to fetch the resources from remote servers when uBlock launches. Delay update to some time after launch.

Negative consequence: the filter lists used are likely out of date.

So there is a compromise here. Which is more important, flawless out-of-box experience or up-to-date filter lists at launch. I now lean toward flawless out-of-box experience.

**Comment by 0xBRM** [comment]
The latter. The filters will be updated in the same session and it's unlikely the user will stumble upon a webpage that features content the previous versions (from the hour/day before) of the lists can't deal with.
